src/main/java/seedu/address/model/events/EventList.java

```diff
@@ -24,6 +25,19 @@ public EventList(ReadOnlyEventList eventList) {
         this.resetData(eventList);
     }
 
+    /**
+     * Returns a deep-copy of an EventList.
+     *
+     * @param oldEventList the EventList to deep-copy.
+     */
+    public EventList(EventList oldEventList) {
+        this();
```

Comment:
What is this `this()` for?

Reply:
Because there is a variable `ObservableList<EventSource> list` belonging to `EventList` class that is initialized in the "default" constructor public EventList(ReadOnlyEventList eventList). 
To avoid duplicating the same code in this constructor (to initialize that list variable), I make use of the default constructor by calling `this()`.